### PR TITLE
Miscellaneous fixes

### PR DESCRIPTION
**Pull request by varunagrawal**
1. Suppress warnings for clang as well as gcc.
2. Fix warning for potentially unassigned pointer.
3. Plotting code fixes.

### CODE REVIEW

File: gtsam/CMakeLists.txt [hunk 1]
```diff
@@ -199,7 +199,7 @@ if(WIN32)
 else()
   if("${CMAKE_BUILD_TYPE}" STREQUAL "Release")
     # Suppress all warnings from 3rd party sources.
-    set_source_files_properties(${3rdparty_srcs} PROPERTIES COMPILE_FLAGS "-w")
+    set_source_files_properties(${3rdparty_srcs} PROPERTIES COMPILE_FLAGS "-w -Wno-everything")
```
Comment (dellaert) [inline gtsam/CMakeLists.txt:202]: what's this?

File: python/gtsam/utils/plot.py [hunk 2]
```diff
@@ -36,18 +36,15 @@ def set_axes_equal(fignum):
     ax.set_zlim3d([origin[2] - radius, origin[2] + radius])
 
 
-def ellipsoid(xc, yc, zc, rx, ry, rz, n):
+def ellipsoid(rx, ry, rz, n):
```
Comment (dellaert) [inline python/gtsam/utils/plot.py:39]: API change? Why?

File: gtsam/CMakeLists.txt (hunk 1 above)
Comment (varunagrawal) [reply gtsam/CMakeLists.txt:202]: That's the flag to suppress warnings in Clang. `-w` already does so for GCC, but Clang still throws warnings in Release mode.

File: python/gtsam/utils/plot.py (hunk 2 above)
Comment (varunagrawal) [reply python/gtsam/utils/plot.py:39]: The center point values are not used in the function anywhere. They are simply wasteful args that serve no purpose.

**Review by dellaert — APPROVED** [review]

**Pull request merged**